src/bundle/Resources/config/services.yaml

```diff
@@ -109,8 +106,45 @@ services:
             - { name: "support_tools.system_info.collector", identifier: "symfony_kernel" }
 
     # SystemInfoOutputFormats
-
     support_tools.system_info.output_format.json:
         class: "%support_tools.system_info.output_format.json.class%"
         tags:
             - { name: "support_tools.system_info.output_format", format: "json" }
+
+    # Gateways
+    EzSystems\EzSupportTools\Storage\MetricsProvider:
+        alias: EzSystems\EzSupportTools\Storage\AggregateMetricsProvider
+
+    EzSystems\EzSupportTools\Storage\AggregateMetricsProvider:
+        arguments:
+            $metrics: !tagged_iterator { tag: ibexa.system_info.metrics, index_by: identifier }
```

Comment:
Just a note that `!tagged_locator` / `ServiceLocator` can be used here instead, since we're looking for a particular service anyway each time, based on identifier.

Reply:
> Just a note that `!tagged_locator` / `ServiceLocator` can be used here instead, since we're looking for a particular service anyway each time, based on identifier.

So something like that: 733f3b8f? If so then maybe you've just found one useful application of locators in favor of iterators. I don't see even static analysis complain on return type, which is good.

Reply:
Pretty much yeah.

This has the added benefit of **not** initializing services in this locator until they are actually used.